toolsrc/src/vcpkg/commands.porthistory.cpp

```diff
@@ -241,7 +215,24 @@ namespace vcpkg::Commands::PortHistory
             root.insert("versions", versions_json);
 
             auto json_string = Json::stringify(root, vcpkg::Json::JsonStyle::with_spaces(2));
-            System::printf("%s\n", json_string);
+
+            if (maybe_output_file.has_value())
+            {
+                auto output_file_path = fs::u8path(maybe_output_file.value_or_exit(VCPKG_LINE_INFO));
+                auto& fs = paths.get_filesystem();
+                std::error_code ec;
+                fs.write_contents(output_file_path, json_string, ec);
```

Comment:
```suggestion
                fs.write_contents(output_file_path, json_string, VCPKG_LINE_INFO);
```
`write_contents` should do the right thing if passed VCPKG_LINE_INFO.